### ISSUE TITLE
make gam available as a Homebrew package ???

### ISSUE THREAD

**Issue by Luke-Jaeger** [GAM-team/GAM#1660]
**any chance that gam could be in the homebrew repositories?**
It would be delightful to just type `brew upgrade gam` and be done

**gam is awesome by the way**
I manage groups & lists for a college CS department and it's so nice to have this tool! Especially since I always have to update the lists at the busiest time of year.

**Comment by Luke-Jaeger** [comment]
> Just type the following: bash <(curl -s -S -L https://gam-shortn.appspot.com/gam-install) -l

I know, it's just nice to have one standard way of updating all my dozens of 3rd-party command line tools

**Comment by chris-zenfolio** [comment]
As a macOS user, the first thing I look for are Homebrew versions of utilities because they are easier to install and maintain.  While I know the current install of gam is only 1 line, it is far easier for me to remember `brew install gam` over the current one.  I can also tell brew to update by simply saying `brew update gam`.  It's a standardized syntax.

As @daethnir mentioned, the bar is higher to get code accepted into Homebrew versus just publishing it as is being done now.  What is `appspot.com`, anyway?  How does an admin go about validating install links, especially if they're inexperienced?  The point of the previous questions is to illustrate that it is easier to use and trust a tool like Homebrew because there is some validation involved.

Please add Homebrew support!

**Comment by dotone-101** [comment]
MacBook-Pro ~ % bash <(curl -s -S -L https://gam-shortn.appspot.com/gam-install)
curl: (7) Failed to connect to raw.githubusercontent.com port 443: Operation timed out

**Comment by jay0lee** [comment]
PRs welcome. We have a *lot* of requests for adding GAM to <yet another packaging system on OS that doesn't actually have native packaging solved> FWIW, I'll never understand how it is Microsoft and Apple have been building OSes for half a decade and still haven't figured out a reliable solution to packaging / installers.

Anyway, PRs are always welcome here if GAM needs some code to integrate or just submit GAM to said packaging system and feel free to let community know about it.

**Comment by Zircoz** [comment]
Imma investigate what is needed for homebrew & hopefully land the ship safely onshore :) feel free to assign this one to me @jay0lee  / @taers232c

**Comment by Zircoz** [comment]
https://github.com/Zircoz/homebrew-GAM

did it :)

**Comment by jay0lee** [comment]
Nice. A couple things 1) it's "GAM7" to disambiguate from gam-adv, standard
gam, etc. 2) to be official, this would need to run as a GitHub Action that
kicks off whenever Ross or I do a release/tag. See how it's being done for
PyPi here:


https://github.com/GAM-team/GAM/blob/main/.github%2Fworkflows%2Fpypi.yml

**Comment by trickfilm** [comment]
thank you !!!!!!

**Comment by Zircoz** [comment]
Hi  @jay0lee !
I created a new PR #1877  to enhance how homebrew package is updated on new releases (event triggered vs polling). I've also added you as colab on homebrew repo; 
We'd also need to set the secret HOMEBREW_TAP_TOKEN on this repo (a fine grained PAT w access to https://github.com/Zircoz/homebrew-GAM ). LMK if you'd like to share the token I created or you want to create it yourself (I'm personally okay with either).

**Comment by jay0lee** [comment]
Heads up that I haven't forgotten about this or the pull request but I think it will be much cleaner to submit the Python source releases to brew core so a command like:
```
brew install gam
```
will just work. I've begun work on that now and will provide an update here when that's done.

do note that I believe that will depend on #1888 being implemented so brew can test with a simple `gam version` command and not get any error or warnings.

**Comment by jay0lee** [comment]
Also I should mention I went back to the package being just "gam" because brew really wants the package name and resultant command from the install to fully align. Calling the pcakage "gam7" and the command "gam" would likely introduce friciton on brew's review process.

**Comment by Zircoz** [comment]
> Heads up that I haven't forgotten about this or the pull request but I think it will be much cleaner to submit the Python source releases to brew core so a command like:
> 
> ```
> brew install gam
> ```
> 
> will just work. I've begun work on that now and will provide an update here when that's done.
> 
> do note that I believe that will depend on [#1888](https://github.com/GAM-team/GAM/issues/1888) being implemented so brew can test with a simple `gam version` command and not get any error or warnings.

Pls feel free to rope me into stuff if you need an extra hand, I love doing mercenary stuff (oss or not) :)
